### PR TITLE
fix(tabs-provider): introduce provider priorities

### LINKED ISSUE #2158: Wrong order of "Create BPMN Diagram (Zeebe)" and "Create BPMN Diagram (Camunda Engine)"
__Describe the Bug__

In the current nightly (16.03.2021), the order of the "Create ..." options is as follows: 
image

However, `Camunda Engine` shall be the most prominent option on top.

__Steps to Reproduce__

1. Use Camunda Modeler Nightly as of 16.03.2021 
2. Click on "create diagram"
3. Wrong order appears

__Expected Behavior__

The expected order is:
```
Create BPMN Diagram (Camuda Engine)
Create BPMN Diagram (Zeebe)
Create DMN Diagram ....
Create Form ...
```

__Environment__

- OS: Manjaro Linux
- Camunda Modeler Version: nightly as of
- Found in the context of https://github.com/camunda/camunda-docs-manual/pull/889

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Try to open file with higher prioritized providers first

Closes #2158

----

Note: I investigated whether it would be possible to create a `before` /  `after` for the menu items like [Electron is already supporting it](https://www.electronjs.org/docs/api/menu-item). This way we could define that, as an example, the Cloud BPMN menu entries will be rendered after the BPMN in any case.

That would work fine (and out of the box) for the `getNewFileMenu` methods of the providers. However, for the `getNewFileButton` we would need to implement [a similar logic](https://github.com/electron/electron/blob/master/lib/browser/api/menu-utils.ts#L96) on our own for the `DropdownButton`. I gave it a chance in a time-boxed manner and did not end up with a reasonable solution. 

That's why I decided to move away from the `before` / `after` -  or a dedicated `priority` -strategy for the menu items (cf. https://github.com/camunda/camunda-modeler/issues/2158#issuecomment-802753760) and went the simplest road, by giving the providers themselves a `priority` and only handle those when trying to open the files (cf. https://github.com/camunda/camunda-modeler/issues/2158#issuecomment-802715108).

This way we keep the existing behavior for the menu items (first come, first serve, btw. a good topic for the Hour of Code).

### CODE REVIEW

File: client/src/app/TabsProvider.js [hunk 1]
```diff
@@ -88,6 +96,51 @@ export default class TabsProvider {
         },
         getNewFileButton() {
           return null;
+        },
+      },
```
Comment (nikku) [inline client/src/app/TabsProvider.js:100]: A suggestion: We could write a test that verifies the correct menu item order is show.

We could also attach the priority to the menu item _only_, as this is the place where it matters at the moment, as far as I understand?

Comment (pinussilvestrus) [reply client/src/app/TabsProvider.js:100]: +1 for the test case

The priority is _not_ for the menu item, but for which provider will be tried to open the file first (https://github.com/camunda/camunda-modeler/pull/2167/files#diff-a8ee53f63c47900983b24616f87f9b04023e1b48be19fd3ee726d0138276b811R493).

I tried to use it for the menu item only, which was more difficult as expected (https://github.com/camunda/camunda-modeler/pull/2167#issue-596577291). That's why I didn't do so.

Comment (nikku) [reply client/src/app/TabsProvider.js:100]: Fine as long as we have a test case that asserts the correct behavior.

Comment (nikku) [reply client/src/app/TabsProvider.js:100]: > The priority is not for the menu item, but for which provider will be tried to open the file first (https://github.com/camunda/camunda-modeler/pull/2167/files#diff-a8ee53f63c47900983b24616f87f9b04023e1b48be19fd3ee726d0138276b811R493).

:+1: on this one.

:+1: on not having a test case, too. A proper test case would require a substatntial amount of work and getting it _stable_ would require introduction of new APIs, too. Let's not do that for now :wink:

Comment (pinussilvestrus) [reply client/src/app/TabsProvider.js:100]: ❤️

**Pull request merged**